### PR TITLE
Use a dedicated thread pool to restart nodes

### LINKED ISSUE #109792: [CI] SnapshotStressTestsIT testRandomActivities failing
**Build scan:**
https://gradle-enterprise.elastic.co/s/w3j3jcaljzhnc/tests/:server:internalClusterTest/org.elasticsearch.snapshots.SnapshotStressTestsIT/testRandomActivities

**Reproduction line:**
```
./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.snapshots.SnapshotStressTestsIT.testRandomActivities" -Dtests.seed=1226FEEF453AB964 -Dtests.locale=de-LU -Dtests.timezone=Antarctica/Vostok -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
[Failure dashboard for `org.elasticsearch.snapshots.SnapshotStressTestsIT#testRandomActivities`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testRandomActivities'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.snapshots.SnapshotStressTestsIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure excerpt:**
```
java.lang.Exception: Test abandoned because suite timeout was reached.

  at __randomizedtesting.SeedInfo.seed([1226FEEF453AB964]:0)

```

### PR DESCRIPTION
**Pull request by nicktindall**
Closes #109792

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotStressTestsIT.java [hunk 1]
```diff
@@ -216,11 +216,13 @@ private static class TrackedCluster {
...
         static final Logger logger = LogManager.getLogger(TrackedCluster.class);
         static final String CLIENT = "client";
+        static final String NODE_RESTARTER = "node_restarter";
 
         private final ThreadPool threadPool = new TestThreadPool(
             "TrackedCluster",
             // a single thread for "client" activities, to limit the number of activities all starting at once
-            new ScalingExecutorBuilder(CLIENT, 1, 1, TimeValue.ZERO, true, CLIENT)
+            new ScalingExecutorBuilder(CLIENT, 1, 1, TimeValue.ZERO, true, CLIENT),
+            new ScalingExecutorBuilder(NODE_RESTARTER, 1, 5, TimeValue.ZERO, true, NODE_RESTARTER)
```
Comment (nicktindall) [inline server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotStressTestsIT.java:225]: It seems very unlikely that there's more than 1 of these running at a time, I made the thread pool max size 5 just incase.

To make it happen even once you need to run with `-Dtests.nightly=true` and run it repeatedly for a while.

Comment (ywangd) [reply server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotStressTestsIT.java:225]: I think there can be 1 to 2 restarter tasks
https://github.com/elastic/elasticsearch/blob/05f80ef058327c1a3d8a911b4872022cb3ff3210/server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotStressTestsIT.java#L284

**Review by ywangd — APPROVED** [review]
LGTM

Thanks for working on this.

File: server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotStressTestsIT.java (hunk 1 above)
Comment (nicktindall) [reply server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotStressTestsIT.java:225]: I might just leave at 5, seeing as it's just a max anyway?

**Pull request merged**